src/utilities/record-utils.ts

```diff
@@ -21,6 +46,21 @@ const RecordUtils = {
             : new record(maybeRecord);
     },
 
+    /**
+     * Function to ensure a given array of objects are instances of a specific Record `T`. If they
+     * are not, one will be instantiated with the given constructor.
+     *
+     * If `maybeRecord` is already an instance of `T`, it returns that value.
+     *
+     * @param maybeRecords Array of objects or Records to be coalesced into Records.
+     * @param record Type of the Record to be instantiated
+     */
+    ensureRecords<T>(maybeRecords: any[], record: Constructor<T>): T[] {
```

Comment:
Thought: What if this was more strongly typed?  Something like this: 
```
ensureRecords<TValue, TRecord extends TValue>(
        maybeRecords: TValue[],
        record: Constructor<TRecord>
    ): TRecord[]
```

Reply:
I don't see why it couldn't be - let me try it out 👍